cmd/publish/main.go

```diff
@@ -105,11 +116,46 @@ type MessagesContext struct {
 	CliClient   *cliClient.CliClient
 }
 
+//go:embed regoDefinitionSchema.yaml
+var regoDefinitionSchemaStr string
+
 func publish(ctx *PublishCommandContext, path string, localConfigContent *localConfig.LocalConfig) (*cliClient.PublishFailedResponse, error) {
 	policiesConfiguration, err := ctx.FilesExtractor.ExtractYamlFileToUnknownStruct(path)
 	if err != nil {
 		return nil, err
 	}
 
+	if customRules, ok := policiesConfiguration["customRules"]; ok {
+		valid, err :=
+			validateIfRegoRules(ctx, customRules.([]interface{}))
+		if err != nil {
+			return nil, err
+		}
+		if !valid {
+			return nil, fmt.Errorf("failed to validate rego custom rules")
+		}
+	}
+
 	return ctx.PublishCliClient.PublishPolicies(policiesConfiguration, localConfigContent.Token)
 }
+
+func validateIfRegoRules(ctx *PublishCommandContext, customRules []interface{}) (bool, error) {
+	for _, regoRule := range customRules {
```

Comment:
at this stage we don't know yet if it's a regoRule
```suggestion
	for _, cusotmRule := range customRules {
```